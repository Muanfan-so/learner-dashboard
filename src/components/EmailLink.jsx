import React from 'react';
import PropTypes from 'prop-types';

import { MailtoLink } from '@edx/paragon';

export const EmailLink = ({ address }) => address && (
  <MailtoLink to={address}>{address}</MailtoLink>
);
EmailLink.defaultProps = { address: null };
EmailLink.propTypes = { address: PropTypes.string };

export default EmailLink;
