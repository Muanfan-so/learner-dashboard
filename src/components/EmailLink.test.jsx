import { shallow } from 'enzyme';

import EmailLink from './EmailLink';

describe('EmailLink', () => {
  it('renders null when no address is provided', () => {
    const wrapper = shallow(<EmailLink />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.isEmptyRender()).toEqual(true);
  });
  it('renders a MailtoLink when an address is provided', () => {
    const wrapper = shallow(<EmailLink address="test@email.com" />);
    expect(wrapper.find('MailtoLink').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
