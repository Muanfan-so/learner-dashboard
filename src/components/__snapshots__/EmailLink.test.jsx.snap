// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailLink renders a MailtoLink when an address is provided 1`] = `
<MailtoLink
  to="test@email.com"
>
  test@email.com
</MailtoLink>
`;

exports[`EmailLink renders null when no address is provided 1`] = `""`;
